### PR TITLE
Fix incorrect failed shards count in APIs for current snapshots

### PR DESCRIPTION
**Pull request by original-brownbear**
The failed shards count approach is wrong, if we have an in-progress snapshot
the difference in counts does not work out to the failed shards count.
Since every failed shard has an entry in the failure list anyway we can just
return its size instead for the correct answer.

### CODE REVIEW

**Review by tlrx — APPROVED** [review]
LGTM, good catch

**Review by fcofdez — APPROVED** [review]
LGTM 👍

File: server/src/internalClusterTest/java/org/elasticsearch/snapshots/GetSnapshotsIT.java [hunk 1]
```diff
@@ -191,6 +191,13 @@ public void testSortAndPaginateWithInProgress() throws Exception {
         assertStablePagination(repos, allSnapshotNames, GetSnapshotsRequest.SortBy.START_TIME);
         assertStablePagination(repos, allSnapshotNames, GetSnapshotsRequest.SortBy.NAME);
         assertStablePagination(repos, allSnapshotNames, GetSnapshotsRequest.SortBy.INDICES);
+        final List<SnapshotInfo> currentSnapshots = clusterAdmin().prepareGetSnapshots(matchAllPattern())
+            .setSnapshots(GetSnapshotsRequest.CURRENT_SNAPSHOT)
+            .get()
+            .getSnapshots();
+        for (SnapshotInfo currentSnapshot : currentSnapshots) {
+            assertThat(currentSnapshot.toString(), currentSnapshot.failedShards(), is(0));
```
Comment (idegtiarenko) [inline server/src/internalClusterTest/java/org/elasticsearch/snapshots/GetSnapshotsIT.java:199]: Should the marcher be `equalTo(0)`?

**Review by idegtiarenko — APPROVED** [review]

File: server/src/internalClusterTest/java/org/elasticsearch/snapshots/GetSnapshotsIT.java (hunk 1 above)
Comment (original-brownbear) [reply server/src/internalClusterTest/java/org/elasticsearch/snapshots/GetSnapshotsIT.java:199]: We used `is(` for all other tests here so I guess we have to go with this for consistency

**Comment by original-brownbear** [comment]
Thanks all!

**Pull request merged**